h2o-py/tests/pyunit_utils/utilsPY.py

```diff
@@ -3643,6 +3643,20 @@ def build_save_model_GLM(params, x, train, respName):
     model.download_mojo(path=TMPDIR)    # save mojo
     return model
 
+def build_save_model_GAM(params, x, train, respName):
+    # build a model
+    model = H2OGeneralizedAdditiveEstimator(**params)
+    model.train(x=x, y=respName, training_frame=train)
+    # save model
+    regex = re.compile("[+\\-* !@#$%^&()={}\\[\\]|;:'\"<>,.?/]")
+    MOJONAME = regex.sub("_", model._id)
+
+    print("Downloading Java prediction model code from H2O")
+    TMPDIR = os.path.normpath(os.path.join(os.path.dirname(os.path.realpath('__file__')), "..", "results", MOJONAME))
```

Comment:
You can just `import tempfile` and then do `tempfile.mkdtemp()` - Please look at `pyunit_generic_model_mojo_glm.py` to see how I did it. It's also guaranteed the file will be removed afterwards, now it's not guaranteed (at least I can't see it). It's a simplification, so I suggest doing it this way.

Reply:
yes, I switched to tempfile